### PR TITLE
Updated moiety code with built-in graph functions

### PR DESCRIPTION
**Pull request by huldash**
Faster without dependency on Bioinformatics Toolbox.

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [] with [X] to check the box)*

### CODE REVIEW

**Comment by laurentheirendt** [comment]
@huldash, eventually, for this error, we might need to do a version check in the function:
```
17:00:21 ================================================================================
17:00:21 Error occurred in testMoieties/testMoieties and it did not run to completion.
17:00:21 
17:00:21     --------------
17:00:21     Error Details:
17:00:21     --------------
17:00:21     Undefined function 'graph' for input arguments of type 'double'.
17:00:21     
17:00:21 ================================================================================
```

File: src/analysis/topology/conservedMoieties/identifyConservedMoieties.m [hunk 1]
```diff
@@ -59,55 +59,44 @@
...
-    % components = connectedComponents(adj); % Matlab implementation of Tarjan's algorithm. Does not work for large networks due to Matlab limitations on stack size.
-    components = find_conn_comp(adj);
-    nComps = length(components);
-end
+% Find connected components of underlying undirected graph.
+% Each component corresponds to an "atom conservation relation".
+[h,~] = find(A == -1);
+[t,~] = find(A == 1);
+adj = sparse([t;h],[h;t],ones(size([t;h]))); % Convert incidence matrix to adjacency matrix
+components = conncomp(graph(adj));
```
Comment (tpfau) [inline src/analysis/topology/conservedMoieties/identifyConservedMoieties.m]: Just a comment:
This function was only introduced in Matlab 2015b and wont work with previous versions.

File: src/analysis/topology/conservedMoieties/identifyConservedMoieties.m [hunk 2]
```diff
@@ -119,67 +108,110 @@
...
+            
+            [g2,o2n2,n2o2] = unique(mgraph2','rows','stable');
+            g2 = g2';
+            urxns2 = rxns2(o2n2);
+            utrans2 = trans2(o2n2);
+            [h2,~] = find(g2 < 0);
+            [t2,~] = find(g2 >0);
+            edges2 = table([h2 t2],urxns2,utrans2,'VariableNames',{'EndNodes' 'Rxn' 'Transition'});
+            nodes2 = table(mets2,comp2,'VariableNames',{'Met' 'Atom'});
+            d2 = digraph(edges2,nodes2);
```
Comment (tpfau) [inline src/analysis/topology/conservedMoieties/identifyConservedMoieties.m]: Similar comment here: `isomorphiosm` was only introduced in 2015b and won't work on previous versions.

File: src/analysis/topology/conservedMoieties/identifyConservedMoieties.m [hunk 3]
```diff
@@ -119,67 +108,110 @@
...
+            
+            % find isomorphism that conserves metabolite and reaction
+            % attributes of nodes and edges
+            p = isomorphism(d1,d2,'NodeVariables','Met','EdgeVariables','Rxn');
```
Comment (tpfau) [inline src/analysis/topology/conservedMoieties/identifyConservedMoieties.m]: Similar comment here: `isomorphiosm` was only introduced in 2016b and won't work on previous versions.

**Pull request merged**